### PR TITLE
Implement light client protocol

### PR DESCRIPTION
**Pull request by terencechain**
This PR implements light client protocol referenced in [eth2.0-specs](https://github.com/ethereum/eth2.0-specs/tree/dev/specs/light_client) using commit [1e74cf5f](https://github.com/ethereum/eth2.0-specs/blob/1e74cf5f0d2866d1811b5abb8548452c3db5be26/specs/light_client/sync_protocol.md)

Change list:
- Add light client containers
- Add light client helpers
- Add light client upgrade handlers

### CODE REVIEW

File: beacon-chain/core/state/BUILD.bazel [hunk 1]
```diff
@@ -50,7 +50,7 @@ go_test(
         "//shared/params:go_default_library",
         "//shared/testutil:go_default_library",
         "//shared/trieutil:go_default_library",
-        "@com_github_prysmaticlabs_go_bitfield//:go_default_library",
+        "@com_github_prysmaticlabs_go_bitfield//:go_default_library",shared/bytesutil/bytes.go
```
Comment (prestonvanloon) [inline beacon-chain/core/state/BUILD.bazel]: ?

File: proto/light/v1alpha1/types.proto [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+
+message CompactCommittee {
+    repeated bytes public_keys = 1 [(gogoproto.moretags) = "ssz-size:\"48,4096\""];
+    repeated uint64 compact_validators = 2 [(gogoproto.moretags) = "ssz-size:\"4096\""];
+}
+
+message LightClientUpdate {
+    // Shard block root for authenticating signature data
+        bytes shard_block_root = 1 [(gogoproto.moretags) = "ssz-size:\"32\""];
```
Comment (prestonvanloon) [inline proto/light/v1alpha1/types.proto]: please check formatting on this file

**Comment by nisdas** [comment]
whats the status on this ? @terencechain

**Comment by terencechain** [comment]
Closing this as light client spec is getting reworked

**Pull request closed without merging**